pkg/controllers/provisioning/v1alpha1/allocation/suite_test.go

```diff
@@ -67,127 +66,115 @@ var _ = AfterSuite(func() {
 })
 
 var _ = Describe("Allocation", func() {
-	var ns *environment.Namespace
 	var provisioner *v1alpha1.Provisioner
-
 	BeforeEach(func() {
-		var err error
-		ns, err = env.NewNamespace()
-		Expect(err).NotTo(HaveOccurred())
 		provisioner = &v1alpha1.Provisioner{
-			ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName()), Namespace: ns.Name},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      strings.ToLower(randomdata.SillyName()),
+				Namespace: "default",
+			},
 			Spec: v1alpha1.ProvisionerSpec{
 				Cluster: &v1alpha1.ClusterSpec{Name: "test-cluster", Endpoint: "http://test-cluster", CABundle: "dGVzdC1jbHVzdGVyCg=="},
 			},
 		}
 	})
 
 	AfterEach(func() {
-		ExpectCleanedUp(ns.Client)
+		ExpectCleanedUp(env.Client)
 	})
 
 	Context("Reconcilation", func() {
 		It("should provision nodes for unconstrained pods", func() {
 			pods := []*v1.Pod{
 				test.PodWith(test.PodOptions{
-					Namespace:  ns.Name,
 					Conditions: []v1.PodCondition{{Type: v1.PodScheduled, Reason: v1.PodReasonUnschedulable, Status: v1.ConditionFalse}},
 				}),
 				test.PodWith(test.PodOptions{
-					Namespace:  ns.Name,
 					Conditions: []v1.PodCondition{{Type: v1.PodScheduled, Reason: v1.PodReasonUnschedulable, Status: v1.ConditionFalse}},
 				}),
 			}
 			for _, pod := range pods {
-				ExpectCreatedWithStatus(ns.Client, pod)
+				ExpectCreatedWithStatus(env.Client, pod)
 			}
-			ExpectCreated(ns.Client, provisioner)
-			ExpectEventuallyReconciled(ns.Client, provisioner)
+			ExpectCreated(env.Client, provisioner)
+			ExpectEventuallyReconciled(env.Client, provisioner)
 
 			nodes := &v1.NodeList{}
-			Expect(ns.Client.List(context.Background(), nodes)).To(Succeed())
+			Expect(env.Client.List(context.Background(), nodes)).To(Succeed())
 			Expect(len(nodes.Items)).To(Equal(1))
 			for _, object := range pods {
 				pod := v1.Pod{}
-				Expect(ns.Client.Get(context.Background(), client.ObjectKey{Name: object.GetName(), Namespace: object.GetNamespace()}, &pod)).To(Succeed())
+				Expect(env.Client.Get(context.Background(), client.ObjectKey{Name: object.GetName(), Namespace: object.GetNamespace()}, &pod)).To(Succeed())
```

Comment:
You're using `context.Background()` a lot. Consider defining a context.Background() at the top and reusing it

Reply:
I'm wary of this contexts are cancellable, which means that sharing a context between tests may cause odd side effects if we ever choose to test cancellation. I could imagine defining a beforeEach with a context, but I'm not sure that simplifies things.